pyaedt/twinbuilder.py

```diff
@@ -309,5 +312,257 @@ def set_sim_setup_parameter(self, var_str, expression, analysis_name="TR"):
+            Sets the value of the Z (ref) in ohms.
+            The default value is ``50``.
+        state_space_dynamic_link_type : str, optional
+            Q3D state space dynamic link type.
+            Possible options are:
+                - ``S`` for S parameters link type.
+                - ``RLGC`` for RLGC Parameters link type.
+                - ``EQ`` for Equivalent Circuit.
+            The default value is ``RLGC``.
+        component_name : str, optional
+            Component name.
+            If not provided a generic name with prefix "SimpQ3DData" will be given.
+        save_project : bool, optional
+            Save project after use.
+            The default value is ``True``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+        >>> oComponentManager.AddDynamicNPortData
+
+        Examples
+        --------
+        Create an instance of Twin Builder.
+
+        >>> from pyaedt import TwinBuilder
+        >>> tb = TwinBuilder()
+
+        Add a Q2D dynamic link component.
+
+        >>> tb.add_q3d_dynamic_component("Q2D_ArmouredCableExample",
+        ...                             "2D_Extractor_Cable",
+        ...                             "MySetupAuto",
+        ...                             "sweep1",
+        ...                             "Original",
+        ...                             "100mm")
+        >>> tb.release_desktop()
+        """
+        dkp = self.desktop_class
+        is_loaded = False
+        app = None
+        if os.path.isfile(source_project):
+            project_path = source_project
+            project_name = os.path.splitext(os.path.basename(source_project))[0]
+            if project_name in dkp.project_list():
+                app = dkp[[project_name, source_design_name]]
+            else:
+                app = dkp.load_project(project_path, source_design_name)
+                is_loaded = True
+        elif source_project in self.project_list:
+            project_name = source_project
+            project_path = os.path.join(self.project_path, project_name + ".aedt")
+            project_path = project_path.replace("/", "\\")
+            app = dkp[[source_project, source_design_name]]
+        else:
+            raise ValueError("Invalid project name or path provided.")
+        if app is None:
+            raise ValueError("Invalid project or design name.")
+        setup = [s for s in app.setups if s.name == setup_name]
+        if not setup:
+            raise ValueError("Invalid setup in selected design.")
+        else:
+            if [sweep for sweep in setup[0].sweeps if sweep.name == sweep_name]:
+                coupling_solution_name = "{} : {}".format(setup_name, sweep_name)
+            else:
+                raise ValueError("Invalid sweep name.")
+        if not [m for m in app.matrices if m.name == coupling_matrix_name]:
```

Comment:
Consider testing this if clause. If it is not possible, then add `# pragma: no cover`